src/ui/actions/PlayerActions.js

```diff
@@ -66,32 +69,66 @@ const start = (_id) => {
     return track._id === trackId;
   });
 
+  // If a track exists
   if (queuePosition > -1) {
     const uri = utils.parseUri(queue[queuePosition].path);
 
     Player.setAudioSrc(uri);
     Player.play();
 
+    let queueCursor = queuePosition; // Clean that variable mess later
+
+    // Check if we have to shuffle the queue
+    if (shuffle) {
+      // need to check that later
+      const index = queue.findIndex((track) => {
+        return track._id === _id;
+      });
+
+      const firstTrack = queue[index];
+
+      queue.splice(_id, 1);
+
+      let m = queue.length;
+      let t;
+      let i;
+      while (m) {
```

Comment:
I wonder if that loop can be expressed in some higher-level construct, like what are we actually doing here?